src/components/containers/Modal.tsx

```diff
@@ -139,6 +146,26 @@ export default function Modal({
     return merge({}, defaultStyle, themeStyle, styleOverrides);
   }, [themeRole, styleOverrides, theme]);
 
+  const TitleComponent =
+    titleSize === "large" ? H3 : titleSize === "medium" ? H4 : H5;
+  const SubtitleComponent =
+    titleSize === "large" ? H4 : titleSize === "medium" ? H5 : H6;
+  const titleTopOffset = subtitle
+    ? titleSize === "large"
+      ? 7
+      : titleSize === "medium"
+      ? 5
+      : 4
+    : titleSize === "large"
+    ? 9
+    : titleSize === "medium"
+    ? 7
+    : 5;
+
+  const headerHeight = title
+    ? titleHeight + (styleOverrides?.header?.compact ? 20 : 40)
```

Comment:
This is when we want the title to use empty space more conservatively. I wasn't sure about this API addition because it's inconsistent with what `styleOverrides` provides right now, but it is a styling thing. What do you think?

Reply:
Ahh I see. Yeah i agree - it doesn't quite fit into what `styleOverrides` provides, but then again i'm not sure about making it as visible as a normal modal prop. I think the regular amount of header empty space is part of the style, so it'd be good to make that a _little_ difficult to employ...

Hmm. But still at the end of the day we should make this all usable and i think it makes more logical sense for `compact` to be a Modal prop, than to be in styleOverrides.

There's my train of thought for today! What do you think?

Reply:
Ok, I'm fine with making it a modal prop for now